x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/subfield_header.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { camelCase, startCase } from 'lodash';
+import { EuiTitle, EuiSpacer } from '@elastic/eui';
+import { fieldToDisplayNameMap } from '../../diff_components/translations';
+import type { DiffableAllFields } from '../../../../../../../common/api/detection_engine';
+
+function shouldDisplaySubfieldLabelsForField(fieldName: string): boolean {
+  return [
+    'data_source',
```

Comment:
Hardcoded field names make it harder to maintain this functionality later on when we decide to remove or rename some fields. The good way is to specify a union of rule field names.

Reply:
Thanks! Updated.